crates/core/src/keyring.rs

```diff
@@ -1,67 +1,65 @@
-//! Module for managing the keyring.
+//! Utilities for interacting with keyring and performing signing operations.
 
 use anyhow::{bail, Context, Result};
 use keyring::Entry;
 pub use keyring::Error;
+use warg_client::RegistryUrl;
 use warg_crypto::signing::PrivateKey;
 
 /// Gets the signing key entry for the given registry and key name.
-pub fn get_signing_key_entry(host: &str, name: &str) -> Result<Entry> {
-    Entry::new(
-        &format!("warg-signing-key:{host}", host = host.to_lowercase()),
-        name,
-    )
-    .context("failed to get keyring entry")
+pub fn get_signing_key_entry(registry_url: &RegistryUrl, key_name: &str) -> Result<Entry> {
+    let label = format!("warg-signing-key:{}", registry_url.safe_label());
+    Entry::new(&label, key_name).context("failed to get keyring entry")
 }
 
-/// Gets the signing key for the given registry host and key name.
-pub fn get_signing_key(host: &str, name: &str) -> Result<PrivateKey> {
-    let entry = get_signing_key_entry(host, name)?;
+/// Gets the signing key for the given registry registry_label and key name.
```

Comment:
Minor nit but it's a littie inconsistent that registry_label is spelled with an underscore and "key name" is not here.